lib/src/line_segment.dart

```diff
@@ -26,112 +25,63 @@ import 'meta.dart';
 /// //addToMap
 /// var addToMap = [polygon, segments]
 
-FeatureCollection<LineString> lineSegment(GeoJSONObject geoJson) {
+FeatureCollection<LineString> lineSegment(GeometryObject geoJson) {
   List<Feature<LineString>> features = [];
-  if (geoJson is! LineString // todo: 2-vertex [LineString]?
-
-      ||
-      geoJson is! MultiLineString ||
-      geoJson is! MultiPolygon) {
-    throw Exception('Type is not supported');
+  if (geoJson is! LineString &&
+      geoJson is! MultiLineString &&
+      geoJson is! MultiPolygon &&
+      geoJson is! Polygon &&
```

Comment:
what about GeometryCollection?

Reply:
can you maybe check for the opposite?

```dart
geoJson is Point || geoJson is Multipoint
```

Reply:
You are right. I will include. 
Another thing about the conditions: since the function does not accept any Point/MultiPoint types I think we will need further check like below, wdyt?:
`FeatureCollection<LineString> lineSegment(GeometryObject geoJson) {
  List<Feature<LineString>> features = [];
  if (geoJson is! LineString &&
      geoJson is! MultiLineString &&
      geoJson is! MultiPolygon &&
      geoJson is! Polygon &&
      geoJson is! Feature &&
      geoJson is! FeatureCollection &&
      geoJson is! GeometryCollection) {
    throw Exception('Unsupported GeoJSONObject type');
  } else if (geoJson is GeometryCollection) {
    for (GeometryType type in geoJson.geometries) {
      if (type is Point || type is MultiPoint) {
        throw Exception('Unsupported GeoJSONObject type');
      }
    }
  } else if (geoJson is FeatureCollection) {
    for (Feature feature in (geoJson as FeatureCollection).features) {
      if (feature.geometry is Point || feature.geometry is MultiPoint) {
        throw Exception('Unsupported GeoJSONObject type');
      }
    }
  } else {
    flattenEach(geoJson,
        (Feature currentFeature, int featureIndex, int multiFeatureIndex) {
      lineSegmentFeature(geoJson, features); //
    });
  }
  return FeatureCollection(features: features);
}`